client/src/Login.tsx

```diff
@@ -1,63 +1,147 @@
+    };
+  }, [])
 
-  const loginUser = () => {
+  const loginUser = (loadConfig : boolean) => {
     //alert('id:' + accessKey + ' secret:' + secretKey);
 
     fetch('http://localhost:3000/authenticate', {
       method: 'GET'
     })
-    .then(response => {
-      //console.log('resp:', resp.json());
-      //console.log('typeof:', typeof response.json());
-      //console.log(response.json());
-      console.log('status:', response.status);
-      if(response.status !== 200) {
-        console.log('bad credentials...');
+      .then(response => {
+        //console.log('resp:', resp.json());
+        //console.log('typeof:', typeof response.json());
+        //console.log(response.json());
+        console.log('status:', response.status);
+        if (response.status !== 200) {
+          console.log('bad credentials...');
+          alert(badCredentials)
+          // Leverage Electron to parse authentication data from local files
+          if(loadConfig) window.electronAPI.getConfig();
+          return;
+        }
+        else {
+          console.log('Valid creds!');
+          navigate('/app');
+          return;
+        }
+      }).catch(err => {
+        alert('Unable to log in. Please ensure the server is running.')
         // Leverage Electron to parse authentication data from local files
-        parseLocalCredentials();
-      }
-      else {
-        console.log('Valid creds!');
-        navigate('/app');
-      }
-    })
+        if(loadConfig) window.electronAPI.getConfig();
+        return;       
+      })
   }
 
-  const buttonPressed = (event : FormEvent) => {
-    event.preventDefault();
+  // Returns true if the input string is empty
+  const emptyString = (input : string ) => {
+    if(input.trim() === '') return true;
+    return false;    
+  }
 
-    // Use the non-empty inputs to configure local files
-    if(accessKey && secretKey) {
+  const buttonPressed = (event: FormEvent) => {
+    event.preventDefault();
 
+    // Don't submit if any of the fields are empty
+    if(emptyString(accessKey) || emptyString(secretKey) || emptyString(clusterName) || emptyString(regionName)) {
+      alert('Please fill out all fields');
+      return;
     }
 
-    loginUser();
+    setLoading(true);
+    // Configure the user's local files with the input fields
+    window.electronAPI.onConfig([accessKey, secretKey, clusterName, regionName]);
+
+    return;
+  }
+
+  const RenderLoadingScreen = () => {
+    if(isLoading) {
+      return (<LoadingScreen type={LoadingScreenType.cyclingStops} />);
+    }
+    return (<></>);
   }
 
   // Need an environmental condition to avoid logging in user if they just logged out
   // Need to render the background with the "loading icon"
-  loginUser();
+  //loginUser();
 
   return (
-    <div id="login" >
+    
+    <div id="login" style={{background: 'url(' + background + ') no-repeat center center fixed',
+    backgroundSize : 'cover'}}>
+      <RenderLoadingScreen />
```

Comment:
Unless there's a good reason, I'm not sure we should have 'Render' in a JSX components name since every JSX component renders something.